### PR TITLE
use copy_op directly insteadof write_scalar

### PR DESCRIPTION
**Pull request by rchaser53**
related: https://github.com/rust-lang/rust/issues/58645

### CODE REVIEW

File: src/intrinsic.rs [hunk 1]
```diff
@@ -50,22 +50,30 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a+'mir>: crate::MiriEvalContextExt<'a,
...
-            "volatile_store" => {
+            "atomic_store_rel" => {
                 let ptr = this.deref_operand(args[0])?;
                 let val = this.read_scalar(args[1])?; // make sure it fits into a scalar; otherwise it cannot be atomic
                 this.write_scalar(val, ptr.into())?;
             }
 
+            "volatile_store" => {
+                let ptr = this.deref_operand(args[0])?;
+                this.copy_op(args[1], ptr.into())?;
```
Comment (RalfJung) [inline src/intrinsic.rs]: This is exactly the same code as `volatile_load`. That doesn't seem right.

File: tests/run-pass/issue-58645.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+fn main() {
+    unsafe {
+        std::ptr::write_volatile(&mut (0, 0), (0, 0));
```
Comment (RalfJung) [inline tests/run-pass/issue-58645.rs]: Please expand the test to write to some place that we can read back from later, and compare the result using `assert_eq!`.

**Comment by rchaser53** [comment]
Thank you the review. I will think and fix it.

**Comment by rchaser53** [comment]
Maybe I fixed it.

File: src/intrinsic.rs (hunk 1 above)
Comment (bjorn3) [reply src/intrinsic.rs]:
```rust
this.copy_op(ptr.into(), dest)?;    // volatile_load
this.copy_op(args[1], ptr.into())?; // volatile_store
```

It is close, but not the same.

File: tests/run-pass/volatile.rs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// related: #58645
+#![feature(core_intrinsics)]
+use std::intrinsics::{volatile_load, volatile_store};
+
+pub fn main() {
+    unsafe {
+        let mut i: isize = 1;
+        volatile_store(&mut i, 2);
+        assert_eq!(volatile_load(&i), 2);
+    }
```
Comment (RalfJung) [inline tests/run-pass/volatile.rs]: Could you add a final `assert_eq!(i, 2);`, just to make sure "normal" loads also see the right thing? Other than that, LGTM.

Comment (RalfJung) [reply tests/run-pass/volatile.rs]: Oh also you changed it back to a type that worked before. Please keep a non-scalar type, like a pair of two integers.

**Comment by rchaser53** [comment]
Oops. Sorry. I changed it back before and add test for 'normal' load.

**Comment by RalfJung** [comment]
Thanks a lot!

**Pull request merged**